### PR TITLE
[MID-78] feat: support otlp http trace exporter & add module version control

### PR DESCRIPTION
**Pull request by jcyamacho**
# feat: support otlp http trace exporter

## Global Features
- add [release process](https://github.com/blacklane/go-libs/blob/MID-78/RELEASING.md)

## Module Features
- (camunda/v2) [_minor_]: update dependencies
- (logger) [_minor_]: update dependencies
- (middleware) [_minor_]: update dependencies
- (tracking) [_minor_]: update dependencies
- (x/events) [_minor_]: update dependencies
- (otel) [**_major_**]: feature update
  - `BREAKING CHANGE:` SetUpOTel now has no endpoint param, instead grpc or
http can be provided as options 
  - `feat:` add http.Client transport to inject otel heades

### CODE REVIEW

**Review by shane-blacklane — APPROVED** [review]
:clap: :rocket:

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Release Process
+
+This project uses the [`multimod` releaser
+tool](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/multimod)
+to manage releases. This document will walk you through how to perform a
+release using this tool for this repository.
+
+## Start a release
+
+First, decide which module sets will have their versions chagned and what those
```
Comment (IvanMartinezBL) [inline RELEASING.md:10]:
```suggestion
First, decide which module sets will have their versions changed and what those
```

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+```
+
+Be sure to update all the appropriate links at the bottom of the file.
+
+Finally, commit this change to your release branch.
+
+### Make a Pull Request
+
+Push your release branch and create a pull request for the changes. Be sure to
+include the curated chagnes your included in the changelog in the description.
```
Comment (IvanMartinezBL) [inline RELEASING.md:60]:
```suggestion
include the curated changes you included in the changelog in the description.
```

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+2. Push tags to the upstream remote. Make sure you
+   push all sub-modules as well.
+
+   ```sh
+   make push-tags VERSION=<version>
+   ```
+
+## Release
+
+Finally create a Release on GitHub. If you are release multiple versions for
```
Comment (IvanMartinezBL) [inline RELEASING.md:94]:
```suggestion
Finally create a Release on GitHub. If you are releasing multiple versions for
```

File: versions.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  xevents:
+    version: v0.2.1
+    modules:
+      - github.com/blacklane/go-libs/x/events
+excluded-modules:
+  - github.com/blacklane/go-libs/internal/tools
+  - github.com/blacklane/go-libs/otel/examples
+  - github.com/blacklane/go-libs/probes/examples
+  - github.com/blacklane/go-libs/x/events/examples/oauth
+  - github.com/blacklane/go-libs/x/events/examples/without-auth
```
Comment (IvanMartinezBL) [inline versions.yaml:35]:
```suggestion
  - github.com/blacklane/go-libs/x/events/examples/without-auth

```

**Review by IvanMartinezBL — CHANGES_REQUESTED** [review]

File: versions.yaml (hunk 4 above)
Comment (jcyamacho) [reply versions.yaml:35]: hi @IvanMartinezBL, thanks a lot for all ur feedback!!! 

In this case I can not see the diff, can u plz help me?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    apk --no-cache add \
+    build-base \
+    git \
+    make
+
+ARG GITHUB_TOKEN
+RUN git config --global url.https://${GITHUB_TOKEN}:@github.com.insteadOf https://github.com
+
+WORKDIR /src
+COPY . ./
```
Comment (shane-blacklane) [inline Dockerfile:14]: newline needed :)

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+$(TOOLS):
+	@mkdir -p $@
+$(TOOLS)/%: | $(TOOLS)
+	cd $(TOOLS_MOD_DIR) && \
+	$(GO) build -o $@ $(PACKAGE)
+
+GOLANGCI_LINT = $(TOOLS)/golangci-lint
+$(GOLANGCI_LINT): PACKAGE=github.com/golangci/golangci-lint/cmd/golangci-lint
+
+MISSPELL = $(TOOLS)/misspell
```
Comment (shane-blacklane) [inline Makefile:26]: nice :+1:

File: versions.yaml (hunk 4 above)
Comment (shane-blacklane) [reply versions.yaml:35]: it's just a newline at the end of file suggestion i think :)

**Review by shane-blacklane — APPROVED** [review]
:rocket: :rocket: :rocket:

File: versions.yaml (hunk 4 above)
Comment (IvanMartinezBL) [reply versions.yaml]: Exactly. Github shows a red icon when the file doesn't end in newline. Just that.

**Review by IvanMartinezBL — APPROVED** [review]

**Review by shane-blacklane — APPROVED** [review]

**Comment by jcyamacho** [comment]
status not updated but drone pipeline is ok: https://drone.blacklane.com/blacklane/go-libs/535

File: otel/http.go [hunk 7]
```diff
@@ -49,3 +49,23 @@ func HTTPInject(ctx context.Context, r *http.Request) {
...
+func (o *otelTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	ctx := req.Context()
+	HTTPInject(ctx, req)
+	return o.Transport.RoundTrip(req)
+}
+
+// InstrumentTransport instruments the given http.RoundTripper with OTel.
+// If the given http.RoundTripper is nil, http.DefaultTransport is used.
+// Requests should be made with the context with the span.
+func InstrumentTransport(t http.RoundTripper) http.RoundTripper {
```
Comment (marcelo-xavier-bl) [inline otel/http.go:66]: Nice one!

**Review by marcelo-xavier-bl — COMMENTED** [review]
Good job, thank you!

**Pull request merged**